### PR TITLE
Fix issue -propagation type is not case insensitive

### PR DESCRIPTION
**Pull request by xyq175com**
Make the context propagation extraction case insensitive. See link:[#3149](https://github.com/open-telemetry/opentelemetry-dotnet/issues/3149)
## Changes
Do a case-insensitive comparison in the Grpc.Core instrumentation

### CODE REVIEW

**Comment by pellared** [comment]
1. Can you mention this in `src/OpenTelemetry.Instrumentation.GrpcCore/CHANGELOG.md`?
2. I think this is about any propagation type - not just B3.
3. Can you try adding a unit test for it to avoid a regression bug?

**Comment by xyq175com** [comment]
> 1. Can you mention this in `src/OpenTelemetry.Instrumentation.GrpcCore/CHANGELOG.md`?
> 2. I think this is about any propagation type - not just B3.
> 3. Can you try adding a unit test for it to avoid a regression bug?

Thank you I updated 'src/OpenTelemetry.Instrumentation.GrpcCore/CHANGELOG.md' and also tested the unit testing for this file and no failure.
image

**Comment by pellared** [comment]
>  also tested the unit testing for this file

I have written "try adding a new unit test" 😉

File: src/OpenTelemetry.Instrumentation.GrpcCore/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # Changelog
 
 ## Unreleased
+  
+## 1.0.0-beta6
+
+* Fix issue -propagation type is not case insensitive
```
Comment (pellared) [inline src/OpenTelemetry.Instrumentation.GrpcCore/CHANGELOG.md:7]:
```suggestion
* Make the context propagation extraction case insensitive.
```

**Review by pellared — APPROVED** [review]

**Comment by pcwiese** [comment]
Looks fine to me. I don't have the power to approve directly on the issue
it seems.

**Review by utpilla — APPROVED** [review]

**Comment by utpilla** [comment]
@xyq175com Could you please update the branch with the latest changes from main?

**Comment by xyq175com** [comment]
@pcwiese I think you have the permission to approve the PR, please help on this, thank you.

**Comment by utpilla** [comment]
@xyq175com Please follow the steps mentioned here if you want a new version of this package released: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#how-to-request-for-release-of-package

**Pull request merged**